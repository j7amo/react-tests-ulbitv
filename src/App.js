// import {useEffect, useState} from "react";
import AppRouter from "./router/AppRouter";
import Navbar from "./components/Navbar/Navbar";

const App =() => {

    // const [data, setData] = useState(null);
    // const [toggle, setToggle] = useState(false);
    // const [value, setValue] = useState('');
    //
    // const onClick = () => {
    //     setToggle(prevToggle => !prevToggle);
    // }
    //
    // const onChange = (evt) => {
    //     setValue(evt.target.value);
    // }
    //
    // useEffect(() => {
    //     setTimeout(() => {
    //         setData({});
    //     }, 100);
    // }, []);
    //
    // return (
    //     <div>
    //         {toggle && <div data-testid='toggle-element'>toggle</div>}
    //         {data && <div>data</div>}
    //         <h1>Hello World</h1>
    //         <h2 data-testid='input-value'>{value}</h2>
    //         <input type='text' placeholder='input value...' value={value} onChange={onChange}/>
    //         <button data-testid='toggle-button' onClick={onClick}>click me</button>
    //     </div>
    // );
    return (
        <>
            <Navbar />
            <AppRouter />
        </>
)
}
export default App;
