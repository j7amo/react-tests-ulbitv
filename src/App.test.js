import { render, screen } from '@testing-library/react';
import App from './App';
import userEvent from "@testing-library/user-event";

describe('App', () => {
  it('renders content', () => {
    render(<App />);
    const headerElement = screen.getByText(/Hello World/i);
    const buttonElement  =screen.getByRole('button');
    const inputElement = screen.getByPlaceholderText(/input value.../i);
    expect(headerElement).toBeInTheDocument();
    expect(buttonElement).toBeInTheDocument();
    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toMatchSnapshot();
  });

  it('doesn\'t render content', () => {
    render(<App />);
    const headerElement = screen.queryByText(/data/i);
    expect(headerElement).toBeNull();
  });

  it('renders data div', async () => {
    render(<App />);
    const headerElement = await screen.findByText(/data/i);
    expect(headerElement).toBeInTheDocument();
  });

  it('toggles content', async () => {
    render(<App />);
    const buttonElement = screen.getByTestId('toggle-button');
    expect(screen.queryByTestId('toggle-element')).toBeNull();
    userEvent.click(buttonElement);
    expect(screen.getByTestId('toggle-element')).toBeInTheDocument();
    userEvent.click(buttonElement);
    expect(screen.queryByTestId('toggle-element')).toBeNull();
  });

  it('correctly renders input', async () => {
    render(<App />);
    const inputElement = screen.getByPlaceholderText(/input value.../i);
    const inputValueHeader = screen.getByTestId('input-value');
    expect(inputValueHeader).toHaveTextContent('');
    userEvent.type(inputElement, 'Hello, MazzaFakka!');
    expect(inputValueHeader).toHaveTextContent('Hello, MazzaFakka!');
  });
});

