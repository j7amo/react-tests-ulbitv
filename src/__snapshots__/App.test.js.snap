// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders "Hello World" elements 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;

exports[`App renders content 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;

exports[`renders "Hello World" header 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;
